Automattic/wp-calypso
# Update jetpack menu functionality and markup

## Proposed Changes

* Move several blocks of code in the jetpack masterbar to their own components
* Remove the use of query and element selectors and instead use React hooks to achieve the functionality we need

## Testing Instructions

1. Click on the Calypso Green live link or checkout this branch locally and start your environment
2. Make sure the header looks exactly as it did before
3. On the mobile menu, test the menu button, the product's tab, the back button, and make sure to test tabbing through it all to make sure the menu closes when tabbed through completely
4. On the desktop menu, test out the products tab, the user menu, and also tabbing through it all to make sure it works as expected
5. Try to break the functionality, there was a lot of moving code in this PR so I could have missed something

## Pre-merge Checklist
